File: jsrc/machine/SxxMachine/Int.java
package SxxMachine;

public class Int extends Real {
	// long Num;

	Int(long i) {
		super(i);
	}

	Term Copy(Prolog m, long t) {
		return new Int(this.Num.longValue());
	}
	
	public final int Arity() {
		return Data.INT;
	}

	public boolean isInt() {
		return true;
	}

}

File: src/machine/SxxMachine/Int.cpp
using namespace std;

#include "Int.h"
#include "Prolog.h"
#include "Term.h"
#include "Data.h"

namespace SxxMachine {

	Int::Int(long long i) : Real(i) {
	}

	Term* Int::Copy(Prolog* m, long long t) {
		return new Int(this->Num->longValue());
	}

	int Int::Arity() {
		return Data::INT;
	}

	bool Int::isInt() {
		return true;
	}
}
